src/buildblock/ProjDataInMemory.cxx

```diff
@@ -73,12 +73,18 @@ ProjDataInMemory::ProjDataInMemory(shared_ptr<const ExamInfo> const& exam_info_s
 void
 ProjDataInMemory::create_buffer(const bool initialise_with_0)
 {
+  if (!initialise_with_0) {
+    this->buffer.set_initialise_with_zeros(false);
+    this->buffer.grow(0, this->size_all());
+    return;
+  }
 #if 0  
   float *b = new float[this->size_all()];
   if (initialise_with_0)
     memset(b, 0, this->size_all()*sizeof(float));
   return b;
 #else
+  this->buffer.set_initialise_with_zeros(true);
   this->buffer = Array<1, float>(static_cast<int>(this->size_all()));
 #endif
```

Comment:
```suggestion
    this->buffer.set_initialise_with_zeros(initialise_with_0);
    this->buffer.resize(0, this->size_all()-1);
```
Note the `-1` (STIR gives min/max indices, not like Python)

Reply:
done